src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -95,6 +95,7 @@ internal sealed class ActiveSolutionBoundTracker : IActiveSolutionBoundTracker,
 
         private void GitEventsMonitor_HeadChanged(object sender, EventArgs e)
         {
+            PreSolutionBindingUpdated?.Invoke(this, EventArgs.Empty);
```

Comment:
This change is unrelated to the stateful branch mapper changes. It would have been better to put it in a separate PR.

Reply:
I raised a separate ticket and PR for this issue: #3500 and #3501